server.go

```diff
@@ -980,7 +980,7 @@ func (s *Server) ShardGroups(database string) ([]*ShardGroup, error) {
 
 // CreateShardGroupIfNotExists creates the shard group for a retention policy for the interval a timestamp falls into.
 func (s *Server) CreateShardGroupIfNotExists(database, policy string, timestamp time.Time) error {
-	c := &createShardGroupIfNotExistsCommand{Database: database, Policy: policy, Timestamp: timestamp}
+	c := &createShardGroupIfNotExistsCommand{Database: database, Policy: policy, Timestamp: timestamp, URL: s.URL()}
```

Comment:
Looking at this again, I believe I can rework it to avoid the deadlock w/o adding it to the `CreateShardGroupsIfNotExists` command.  Will fix.